synapse/daemon.py

```diff
@@ -234,7 +234,7 @@ def _onDmonFini(self):
                 if isinstance(share, s_coro.Fini):
                     await share.fini()
 
-            await asyncio.wait(link.fini() for link in self.connectedlinks)
+            await asyncio.wait([link.fini() for link in self.connectedlinks], loop=s_glob.plex.loop)
```

Comment:
technically you don't have to make this a list (a generator will do), but no bigs.

Reply:
So the test fails for me if I leave it as a generator object.

Reply:
Yea, doc seem to say a "sequence of futures" not an "iterable", so it may require the list.

Reply:
actually tracing through the code in stdlib, a generator is considered a coroutine itself (don't ask... ) so we have to have some sort of fixed iterable we hand in here

Reply:
I'm gonna submit that as a bug to Python. It uses an incorrect check